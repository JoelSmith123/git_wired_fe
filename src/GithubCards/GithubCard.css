

.GithubCard {
  width:   90%;
  margin:  2%;
  padding: 2%;
  width:   30%;
  height:  auto;
  background-color: #BDCCD8;

  display:         flex;
  flex-direction:  column;
  flex-wrap:       wrap;
  justify-content: center;
  align-items:     stretch;
  text-align:      center;
}

.GithubCardMinimum {
  display:         flex;
  flex-direction:  row;
  flex-wrap:       wrap;
  justify-content: space-between;
  align-items:     stretch;
  text-align:      center;
}

.GithubCardHeaders {
  width:           60%;
  display:         flex;
  flex-direction:  column;
  flex-wrap:       nowrap;
  justify-content: space-around;
  align-items:     stretch;
  text-align:      left;
}

.GithubCardHeaders div {

}

.GithubCardLabels {
  width:           40%;
  display:         flex;
  flex-direction:  column;
  flex-wrap:       nowrap;
  align-items:     stretch;
  text-align:      center;
  /* font-size:       70%; */
    font-size:     1.2vw;
}

.GithubCardLabels div {
  width:            100%;
  background-color: white;
  padding:          1%;
  margin:           1%;
  text-overflow: ellipsis;
  overflow:      hidden;
  white-space:   nowrap;
}

.CardTitle {
  font-size:   2vw;
  font-weight: bold;
  margin:      1%;
  text-overflow: ellipsis;
  overflow:      hidden;
  white-space:   nowrap;
}

.CardNumber {
  font-size:   1.7vw;
  font-weight: bolder;
}

.CardDescription {
  font-size:        1.7vw;
  margin-top:       3%;
  text-align:       left;
  background-color: lightgrey;
  width:            90%;
  padding:          2%;
  align-self:       center;
}
