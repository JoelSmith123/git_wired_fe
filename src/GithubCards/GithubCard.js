import React, { Component } from 'react';
import './GithubCard.css'


export default class GithubCard extends Component {


  // --- Card Title ---

  getTitle = () => {
    return this.props.card['cardTitle']
  }

  renderTitle = () => {
    return (
      <div className='CardTitle'> { this.getTitle() } </div>
    )
  }

  // --- Card Description ---

  getDescription = () => {
    return this.props.card['cardDescription']
  }

  renderDescription = () => {
    return (
      <div className='CardDescription'> { this.getDescription() } </div>
    )
  }

  // --- Card Number ---

  getCardURL = () => {
    return this.props.card['cardURL']
  }

  getNumber = () => {
    return this.props.card['cardNumber']
  }

  renderNumber = () => {
    return (
      <div className='CardNumber'>
        <a src={ this.getCardURL() }>
          #{ this.getNumber() }
        </a>
      </div>
    )
  }

  // --- Card Status ---

  getStatus = () => {
    return this.props.card['cardStatus']
  }

  renderStatus = () => {
    return (
      <div className='CardStatus'> { this.getStatus() } </div>
    )
  }

  // --- Card Column ---

  getColumn = () => {
    return this.props.card['cardColumn']
  }

  renderColumn = () => {
    return (
      <div className='CardColumn'> { this.getColumn() } </div>
    )
  }


  // --- Card Connection ---

  // TO DO - determines & Displays if card is connected to wireframe element


  // --- SHOW MORE / LESS ---

  //  TO DO - add a button to toggle display of  card description


  render() {
    return (
      <span className='GithubCard'>

        <div className='GithubCardMinimum'>
          <span className='GithubCardHeaders'>
            { this.renderTitle()       }
            { this.renderNumber()      }
          </span>
          <span className='GithubCardLabels'>
            { this.renderStatus()      }
            { this.renderColumn()      }
          </span>
        </div>

        <div className='GithubCardBody'>
          { this.renderDescription() }
        </div>

        <div>TO DO - See More / Less</div>
      </span>
    )
  }


}
