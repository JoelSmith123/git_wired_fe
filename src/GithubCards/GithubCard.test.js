import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import GithubCard from './GithubCard.js'



describe('GithubCard', () => {
  let mockData
  beforeEach(() => {
    mockData = {
       'cardTitle':       'This is the title',
       'cardDescription': 'This is lots of long text from the body of my issue',
       'cardNumber':       13,
       'cardURL':         'https://github.com/JoelSmith123/git_wired_fe/issues/23',
       'cardStatus':      'Open',
     }  
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<GithubCard card={ mockData } />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<GithubCard card={ mockData }/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('gets title from card', () => {
    const wrapper = shallow(<GithubCard card={ mockData }/>)
    
    expect(wrapper.instance().getTitle()).toEqual(mockData.cardTitle)
  })

  it('gets description from card', () => {
    const wrapper = shallow(<GithubCard card={ mockData }/>)

    expect(wrapper.instance().getDescription()).toEqual(mockData.cardDescription)    
  })

  xit('gets card number from card', () => {
    // TO DO - TEST ME
  })

  xit('gets card url from card', () => {
    // TO DO - TEST ME
  })

  xit('gets card status from card', () => {
    // TO DO - TEST ME
  })

});
