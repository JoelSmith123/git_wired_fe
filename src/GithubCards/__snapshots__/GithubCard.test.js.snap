// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubCard should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GithubCard
    card={
      Object {
        "cardDescription": "This is lots of long text from the body of my issue",
        "cardNumber": 13,
        "cardStatus": "Open",
        "cardTitle": "This is the title",
        "cardURL": "https://github.com/JoelSmith123/git_wired_fe/issues/23",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="GithubCardMinimum"
        >
          <span
            className="GithubCardHeaders"
          >
            <div
              className="CardTitle"
            >
              This is the title
            </div>
            <div
              className="CardNumber"
            >
              <a
                href="https://github.com/JoelSmith123/git_wired_fe/issues/23"
              >
                #
                13
              </a>
            </div>
          </span>
          <span
            className="GithubCardLabels"
          >
            <div
              className="CardStatus"
            >
              Open
            </div>
          </span>
        </div>,
        <div
          className="CardDescription"
        >
          This is lots of long text from the body of my issue
        </div>,
      ],
      "className": "GithubCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="GithubCardHeaders"
            >
              <div
                className="CardTitle"
              >
                This is the title
              </div>
              <div
                className="CardNumber"
              >
                <a
                  href="https://github.com/JoelSmith123/git_wired_fe/issues/23"
                >
                  #
                  13
                </a>
              </div>
            </span>,
            <span
              className="GithubCardLabels"
            >
              <div
                className="CardStatus"
              >
                Open
              </div>
            </span>,
          ],
          "className": "GithubCardMinimum",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="CardTitle"
                >
                  This is the title
                </div>,
                <div
                  className="CardNumber"
                >
                  <a
                    href="https://github.com/JoelSmith123/git_wired_fe/issues/23"
                  >
                    #
                    13
                  </a>
                </div>,
              ],
              "className": "GithubCardHeaders",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "This is the title",
                  "className": "CardTitle",
                },
                "ref": null,
                "rendered": "This is the title",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="https://github.com/JoelSmith123/git_wired_fe/issues/23"
                  >
                    #
                    13
                  </a>,
                  "className": "CardNumber",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "#",
                      13,
                    ],
                    "href": "https://github.com/JoelSmith123/git_wired_fe/issues/23",
                  },
                  "ref": null,
                  "rendered": Array [
                    "#",
                    13,
                  ],
                  "type": "a",
                },
                "type": "div",
              },
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="CardStatus"
              >
                Open
              </div>,
              "className": "GithubCardLabels",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Open",
                "className": "CardStatus",
              },
              "ref": null,
              "rendered": "Open",
              "type": "div",
            },
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This is lots of long text from the body of my issue",
          "className": "CardDescription",
        },
        "ref": null,
        "rendered": "This is lots of long text from the body of my issue",
        "type": "div",
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="GithubCardMinimum"
          >
            <span
              className="GithubCardHeaders"
            >
              <div
                className="CardTitle"
              >
                This is the title
              </div>
              <div
                className="CardNumber"
              >
                <a
                  href="https://github.com/JoelSmith123/git_wired_fe/issues/23"
                >
                  #
                  13
                </a>
              </div>
            </span>
            <span
              className="GithubCardLabels"
            >
              <div
                className="CardStatus"
              >
                Open
              </div>
            </span>
          </div>,
          <div
            className="CardDescription"
          >
            This is lots of long text from the body of my issue
          </div>,
        ],
        "className": "GithubCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="GithubCardHeaders"
              >
                <div
                  className="CardTitle"
                >
                  This is the title
                </div>
                <div
                  className="CardNumber"
                >
                  <a
                    href="https://github.com/JoelSmith123/git_wired_fe/issues/23"
                  >
                    #
                    13
                  </a>
                </div>
              </span>,
              <span
                className="GithubCardLabels"
              >
                <div
                  className="CardStatus"
                >
                  Open
                </div>
              </span>,
            ],
            "className": "GithubCardMinimum",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="CardTitle"
                  >
                    This is the title
                  </div>,
                  <div
                    className="CardNumber"
                  >
                    <a
                      href="https://github.com/JoelSmith123/git_wired_fe/issues/23"
                    >
                      #
                      13
                    </a>
                  </div>,
                ],
                "className": "GithubCardHeaders",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "This is the title",
                    "className": "CardTitle",
                  },
                  "ref": null,
                  "rendered": "This is the title",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <a
                      href="https://github.com/JoelSmith123/git_wired_fe/issues/23"
                    >
                      #
                      13
                    </a>,
                    "className": "CardNumber",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "#",
                        13,
                      ],
                      "href": "https://github.com/JoelSmith123/git_wired_fe/issues/23",
                    },
                    "ref": null,
                    "rendered": Array [
                      "#",
                      13,
                    ],
                    "type": "a",
                  },
                  "type": "div",
                },
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="CardStatus"
                >
                  Open
                </div>,
                "className": "GithubCardLabels",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Open",
                  "className": "CardStatus",
                },
                "ref": null,
                "rendered": "Open",
                "type": "div",
              },
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is lots of long text from the body of my issue",
            "className": "CardDescription",
          },
          "ref": null,
          "rendered": "This is lots of long text from the body of my issue",
          "type": "div",
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
